### PR TITLE
Remove regular expressions, speeds up untar

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

regex is very slow for tar file checking

#### Summary of Changes

Use custom loop, speeds up untar by 3x. 15s => 5s.

Fixes #

### CODE REVIEW

File: runtime/src/hardened_unpack.rs [hunk 1]
```diff
@@ -167,24 +166,46 @@ pub fn unpack_snapshot<A: Read, P: AsRef<Path>>(
...
+                saw_numbers = false;
+                periods += 1;
+            } else {
+                return false;
+            }
+        } else {
+            saw_numbers = true;
+        }
+    }
+    saw_numbers
```
Comment (ryoqun) [inline runtime/src/hardened_unpack.rs:196]: maybe this needs to be changed `periods == 1 && saw_numbers`?

I think "232323" incorrectly returns false.

File: runtime/src/hardened_unpack.rs [hunk 2]
```diff
@@ -167,24 +166,46 @@ pub fn unpack_snapshot<A: Read, P: AsRef<Path>>(
     )
 }
 
-fn is_valid_snapshot_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
-    let like_storage = Regex::new(r"^\d+\.\d+$").unwrap();
-    let like_slot = Regex::new(r"^\d+$").unwrap();
```
Comment (ryoqun) [inline runtime/src/hardened_unpack.rs:172]: maybe, these regex isn't jit compiled? unfortunate. xD

**Comment by ryoqun** [comment]
> Use custom loop, speeds up untar by 3x. 15s => 5s.

cool. thanks working on this!

File: runtime/src/hardened_unpack.rs (hunk 1 above)
Comment (sakridge) [reply runtime/src/hardened_unpack.rs]: yes, fixed and added test

File: runtime/src/hardened_unpack.rs (hunk 2 above)
Comment (sakridge) [reply runtime/src/hardened_unpack.rs:172]: Can't say I've ever come across a regex that is truly fast, but maybe :)

**Review by ryoqun — APPROVED** [review]
lgtm!

**Pull request merged**